lib/signed-xml.js

```diff
@@ -861,7 +870,7 @@ SignedXml.prototype.getKeyInfo = function(prefix) {
 
   if (this.keyInfoProvider) {
     res += "<" + currentPrefix + "KeyInfo>"
-    res += this.keyInfoProvider.getKeyInfo(this.signingKey, prefix)
+    res += this.keyInfoProvider.getKeyInfo(this.signingCert, prefix)
```

Comment:
potentially breaking change, since existing users may be providing `signingKey`. Is there a better way?

Reply:
There might be, but right now none of the code in the code-base actually uses that first argument, so what does it matter?

There is only one concrete implementation of `KeyInfoProvider`, which is `FileKeyInfo`, and that doesn't use the first argument at all.

https://github.com/yaronn/xml-crypto/blob/3deacfcd519b5ba97ee67ca5f0046d8cc4a7a78c/lib/signed-xml.js#L20-L24

Reply:
Granted, this PR does change that, but we need to establish how things are supposed to work first, and then we can consider a change. I'm still considering your advice and what the README says about rolling our own instead of enhancing `FileKeyInfo`. See #243 .

Reply:
@cjbarth  - So I agree with your arguments in #243 to add an upstream in-memory `KeyInfoProvider`. What I think we can do to provide some semblance of backward compatibility is to supply `signingKey` as an argument if present, otherwise falling back on `signingCert`. Something like...
```suggestion
    res += this.keyInfoProvider.getKeyInfo(this.signingKey || this.signingCert, prefix)
```

wdyt? If you're happy with this I'm happy for this to rapidly approve, merge and release this PR

Reply:
@LoneRifle I like where you're going with that compromise, but wouldn't it make more sense to "prefer the new way and default to the old way?" (e.g. `this.signingCert || this.signingKey`)

Reply:
Good point!